ui/app/src/components/ItemTypeIcon/ItemTypeIcon.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2007-2021 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import UnknownStateIcon from '@material-ui/core/SvgIcon/SvgIcon';
+import Js from '../Icons/Js';
+import JsonIcon from '../Icons/Json';
+import Groovy from '../Icons/Groovy';
+import Freemarker from '../Icons/Freemarker';
+import Html from '../Icons/Html';
+import Css from '../Icons/Css';
+import ComponentIcon from '../Icons/Component';
+import PageIcon from '../Icons/Page';
+import LevelDescriptorIcon from '../Icons/LevelDescriptor';
+import Tooltip from '@material-ui/core/Tooltip/Tooltip';
+import * as React from 'react';
+import { ItemDisplayProps } from '../ItemDisplay';
+import ImageIcon from '@material-ui/icons/ImageOutlined';
+import CodeRounded from '@material-ui/icons/CodeRounded';
+import FontIcon from '@material-ui/icons/FontDownloadOutlined';
+import TextIcon from '@material-ui/icons/SubjectRounded';
+import FolderIcon from '@material-ui/icons/FolderOpenRounded';
+import TaxonomyIcon from '@material-ui/icons/LocalOfferOutlined';
+import { DetailedItem, SandboxItem } from '../../models/Item';
+import { capitalize } from '../../utils/string';
+
+export interface ItemTypeIconProps {
+  item: DetailedItem | SandboxItem;
+  classes: ItemDisplayProps['classes'];
```

Comment:
Declare class key (even if it's just 1).

```suggestion
  classes?: ItemTypeIconClassKey;
```